homeassistant/components/climate/econet.py

```diff
@@ -0,0 +1,234 @@
+    ATTR_TEMPERATURE)
+from homeassistant.config import load_yaml_config_file
+
+REQUIREMENTS = ['pyeconet==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_VACATION_START = 'next_vacation_start_date'
+ATTR_VACATION_END = 'next_vacation_end_date'
+ATTR_ON_VACATION = 'on_vacation'
+ATTR_TODAYS_ENERGY_USAGE = 'todays_energy_usage'
+ATTR_IN_USE = 'in_use'
+
+ATTR_START_DATE = 'start_date'
+ATTR_END_DATE = 'end_date'
+
+SERVICE_ADD_VACATION = 'econet_add_vacation'
+SERVICE_DELETE_VACATION = 'econet_delete_vacation'
+
+ADD_VACATION_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Optional(ATTR_START_DATE): cv.positive_int,
+    vol.Required(ATTR_END_DATE): cv.positive_int,
+})
+
+DELETE_VACATION_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+DOMAIN = 'econet'
+
+HA_STATE_TO_ECONET = {
+    STATE_ECO: 'Energy Saver',
+    STATE_ELECTRIC: 'Electric',
+    STATE_HEAT_PUMP: 'Heat Pump',
+    STATE_GAS: 'gas',
+    STATE_HIGH_DEMAND: 'High Demand',
+    STATE_OFF: 'Off',
+}
+
+ECONET_STATE_TO_HA = {value: key for key, value in HA_STATE_TO_ECONET.items()}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the EcoNet water heaters."""
+    from pyeconet.api import PyEcoNet
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN]['water_heaters'] = []
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    econet = PyEcoNet(username, password)
+    water_heaters = econet.get_water_heaters()
+    for water_heater in water_heaters:
+        _temp_water_heater = EcoNetWaterHeater(water_heater)
+        add_devices([_temp_water_heater])
+        hass.data[DOMAIN]['water_heaters'].append(_temp_water_heater)
+
+    def service_handle(service):
+        """Handler for services."""
+        entity_ids = service.data.get('entity_id')
+        all_water_heaters = hass.data[DOMAIN]['water_heaters']
+        _water_heaters = []
+        if entity_ids is None:
+            _water_heaters = all_water_heaters
+        else:
+            for _water_heater in all_water_heaters:
+                if _water_heater.entity_id in entity_ids:
+                    _water_heaters.append(_water_heater)
+
+        for _water_heater in _water_heaters:
+            if service.service == SERVICE_ADD_VACATION:
+                start = service.data.get(ATTR_START_DATE)
+                end = service.data.get(ATTR_END_DATE)
+                _water_heater.add_vacation(start, end)
+            if service.service == SERVICE_DELETE_VACATION:
+                # As far as I can tell you can only have 1 vacation at a time
+                # This logic should maybe be moved into pyeconet?
+                try:
+                    _water_heater.water_heater.vacations[0].delete()
```

Comment:
I think I was originally thinking that if there was some way to have more than one this would keep non HA set vacations safe, but that isn't the case. We can just remove them all.